index.html

```diff
@@ -46,40 +46,60 @@ <h3 id="difficulty"></h3>
       fetch("https://opentdb.com/api.php?amount=10&type=multiple")
         .then((response) => response.json())
         .then((json) => {
-          //クイズを表示する
-          function showQuestion(n) {
-            const quizObj = json.results[n];
-            const genreText = quizObj.category;
-            const dificultyText = quizObj.difficulty;
-            const answers = [];
-            const answer = answers.push(quizObj.correct_answer);
-            const choiceAnswers = [...quizObj.incorrect_answers, ...answers];
-            const question = quizObj.question;
-            startBtn.style.display = "none";
-            genre.style.display = "block";
-            difficulty.style.display = "block";
-            title.textContent = "問題" + (n + 1);
-            genre.textContent = "[ジャンル]" + genreText;
-            difficulty.textContent = "[難易度]" + dificultyText;
-            questionText.textContent = question;
-            for (let i = 0; i < choiceAnswers.length; i++) {
-              const choiceBtn = document.createElement("button");
-              choiceBtn.setAttribute("class", "choice-btn " + i);
-              choiceBtn.textContent = choiceAnswers[i];
-              choiceArea.appendChild(choiceBtn);
+          //クラスの設定
+          class Question {
+            constructor(n) {
+              this.dataObj = json.results[n];
+              this.title = `問題 ${n + 1}`;
+              this.genre = `[ジャンル] ${json.results[n].category}`;
+              this.difficultyText = `[難易度] ${json.results[n].difficulty}`;
+              this.text = json.results[n].question;
+            }
+            show() {
+              //クイズを表示させる
+              startBtn.style.display = "none";
+              genre.style.display = "block";
+              difficulty.style.display = "block";
+              title.textContent = this.title;
+              genre.textContent = this.genre;
+              difficulty.textContent = this.difficultyText;
+              questionText.textContent = this.text;
+              const answers = [];
+              const answer = answers.push(this.dataObj.correct_answer);
+              const choiceAnswers = [
+                ...this.dataObj.incorrect_answers,
+                ...answers,
+              ];
+              for (let i = 0; i < choiceAnswers.length; i++) {
+                const choiceBtn = document.createElement("button");
+                choiceBtn.setAttribute("class", "choice-btn " + i);
+                choiceBtn.textContent = choiceAnswers[i];
+                choiceArea.appendChild(choiceBtn);
+              }
             }
           }
 
-          // //開始ボタンクリックで最初のクイズを表示
+          //複数のインスタンス作成し配列としてまとめる
+          function instances() {
+            const Questions = [];
```

Comment:
ローカル変数を `PascalCase` (先頭大文字) で定義することは普通しません。
```suggestion
            const questions = [];
```